pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -225,6 +236,13 @@ def configure_ddp(self):
             device_ids=self.determine_ddp_device_ids(),
             **self._ddp_kwargs,
         )
+        register_ddp_comm_hook(
+            ddp_comm_state=self.ddp_comm_state,
+            ddp_comm_hook=self.ddp_comm_hook,
+            ddp_comm_wrapper=self.ddp_comm_wrapper,
+            model=self._model,
+            is_single_process_single_device=self.is_single_process_single_device,
+        )
```

Comment:
- what happens if the comm state/hook/wrapper aren't set? 
- we should move this to a separate function such that plugins can extend this plugin and customize the comm hooks without needing to change lines 233-238

Reply:
> what happens if the comm state/hook/wrapper aren't set

if `ddp_comm_hook` is not set, there will be no hook registered. state could be None (as we have stateless hook / stateful hook). For wrapper, if not set, then will not apply the wrap to `ddp_comm_hook`.

> we should move this to a separate function
I see, we could have an abstract method defined for base `TrainingTypePlugin` and in `pre_dispatch`, we could call that after `configure_ddp`
https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/plugins/training_type/ddp.py#L243-L252